baybe/kernels/basic.py

```diff
@@ -43,3 +44,24 @@ class MaternKernel(Kernel):
 
     Only takes the values 0.5, 1.5 or 2.5. Larger values yield smoother interpolations.
     """
+
+    lengthscale_prior: Optional[Prior] = field(default=None)
+    """An optional prior on the kernel lengthscale."""
+
+    def to_gpytorch(self, *args, **kwargs):  # noqa: D102
+        # See base class.
+        # Provide the transformed lengthscale as keyword argument.
+        return super().to_gpytorch(
```

Comment:
I know realize the limitations of this approach, because unfortunately the `gpytorch` constructors do not list their arguments explicitly but go via kwargs =/ 

Therefore, I think it's rather dangerous to continue keeping everything implicit.... I think we need to change direction and rather make stuff explicit. But let's discuss together

Reply:
"Let's discuss together" as is "Let's discuss in detail in the next Dev Meeting"?

Reply:
Yes, we could have another meeting e.g. this afternoon or on thursday?

But basically I see the following paths:
1) we keep the current approach, which I consider error-prone due to gpytorch design
2) we augment the current approach to also screen base classes (probably even more error-prone)
3) we keep an explicit list of attributes to be passed (explicit but perhaps not elegant)
4) we drop the matching helper and simply pass all attributes, with the convention that the classes can only contain relevant attributes for gpytorch (currently see no reason against this, but could be mistaken)

Reply:
@Scienfitz @AdrianSosic Ping that variant 2 is now implemented. The code now also explicitly fetches the fields of gpytorch's base `Kernel` class and includes whatever it finds there in the dictionary. This works well, and it is no longer necessary to manually pass the lengthscale. I also verified this manually:
<img width="536" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/cc61aafa-29c2-466b-8db7-6a1b85d4804c">
<img width="670" alt="image" src="https://github.com/emdgroup/baybe/assets/84016733/46272fcd-3387-4030-bd9e-bd7f0c885676">

Reply:
good this looks like the intended use
consider my additional point about including not only the prior but also the initial value into our kernel classes

imo we dont need to argue about whether thats useful or not, these starting values are used in our code, hence I suggest to include them in the grouping of our classes as not doing that seems weird

Reply:
Agree, having the initial value stuff as part of our classes in form of **optional** attributes seems reasonable. Would in fact also help me quite a lot in refactoring the GP part. I suggest the pattern: `*_prior` and `*_initial_value`. What do you think?

Reply:
fine with these names

whats the role of `_initial_value` in the case where there is no prior? is the likelihood estimate also an iterative estimate which needs a starting value or will the starting value be ignored?

Reply:
Yes, in both cases the value is optimized. If no prior is provided, then via regular marginal likelihood, otherwise MAP inference. So I think that would work in both cases. No clue yet what will change once we get to fully Bayesian models, but this decision can wait, I think.

Reply:
Will thus resolve, please reopen if this discussion should continue here